server/cluster/unsafe_recovery_controller_test.go

```diff
@@ -1065,11 +1055,9 @@ func TestTimeout(t *testing.T) {
 
 	time.Sleep(time.Second)
 	req := newStoreHeartbeat(1, nil)
+	req.StoreReport = &pdpb.StoreReport{Step: 1}
 	resp := &pdpb.StoreHeartbeatResponse{}
 	recoveryController.HandleStoreHeartbeat(req, resp)
-	re.Equal(exitForceLeader, recoveryController.GetStage())
```

Comment:
exit force leader is not triggered for timeout?

Reply:
No, because there is no force leader.

Reply:
In the past, the error handling function changes the stage to exit force leader regardless of whether there are force leaders or not. In the new implementation, we only change stage to exit force leader when needed, this is the difference.